### PR TITLE
Add letters list test

### PR DESCRIPTION
**Pull request by cvalarida**
Had to do some fancy footwork to make sure all the checkboxes got unchecked, but it was pretty straightforward otherwise.

### CODE REVIEW

File: test/letters/01-authed.e2e.spec.js [hunk 1]
```diff
@@ -83,6 +83,43 @@ module.exports = E2eHelpers.createE2eTest(
       .click('div.step-content > p:nth-child(3) > a') // link to go back to confirm-address
       .expect.element('.city-state').to.be.present.before(Timeouts.normal);
 
+    // -- Go to letters list -- //
+
+    client
+      .click('.view-letters-button')
+      .assert.urlContains('/letters/letter-list')
+      .waitForElementVisible('.step-content', Timeouts.normal)
+      .click('.step-content div.form-review-panel:nth-of-type(4) button') // open the bsl accordion
```
Comment (markgreenburg) [inline test/letters/01-authed.e2e.spec.js]: 😬

File: test/letters/01-authed.e2e.spec.js [hunk 2]
```diff
@@ -83,6 +83,43 @@ module.exports = E2eHelpers.createE2eTest(
       .click('div.step-content > p:nth-child(3) > a') // link to go back to confirm-address
       .expect.element('.city-state').to.be.present.before(Timeouts.normal);
 
+    // -- Go to letters list -- //
+
+    client
+      .click('.view-letters-button')
+      .assert.urlContains('/letters/letter-list')
```
Comment (markgreenburg) [inline test/letters/01-authed.e2e.spec.js]: Is the new URL available instantly or does there need to be a wait somewhere in there? Not sure how that works with Router.

File: test/letters/01-authed.e2e.spec.js [hunk 3]
```diff
@@ -83,6 +83,43 @@ module.exports = E2eHelpers.createE2eTest(
...
+    // poke all the checkboxes and expect them to all be unselected
+    client
+      .expect.element('#militaryService').to.have.attribute('checked').equals('true');
+    client
+      .click('#militaryService')
+      .expect.element('#militaryService').to.not.have.attribute('checked');
+    client.execute(() => {
+      return Array.from(document.querySelectorAll('#benefitInfoTable input[type="checkbox"]'))
+        .map(e => e.getAttribute('id'));
+    }, [], (obj) => {
```
Comment (markgreenburg) [inline test/letters/01-authed.e2e.spec.js]: Interesting . . . this callback receives an object that includes the value returned from body param function.

**Review by markgreenburg — APPROVED** [review]
:shipit:

File: test/letters/01-authed.e2e.spec.js (hunk 1 above)
Comment (cvalarida) [reply test/letters/01-authed.e2e.spec.js]: I considered dynamically adding an id to these accordion buttons, but it seems like more work than it was worth. I pretty much hate this selector, but it works. Until we change things. Then we cry. 😢

File: test/letters/01-authed.e2e.spec.js (hunk 2 above)
Comment (cvalarida) [reply test/letters/01-authed.e2e.spec.js]: It should be available instantly. If this fails suspiciously later, we can reassess. :man_shrugging:

File: test/letters/01-authed.e2e.spec.js (hunk 3 above)
Comment (cvalarida) [reply test/letters/01-authed.e2e.spec.js]: It's a fun, undocumented feature. 😏 
'Course, I say that, but it probably is documented somewhere. I just had to `console.log` and experiment to figure out how to make it work.

**Pull request merged**